### ISSUE TITLE
Add git-template to project

### ISSUE THREAD

**Issue by lucdekens** [vmware/dscr-for-vmware#26]
Can you please add a git-template to the project?
In the template I would like to have the line

**_Signed-off-by: Your Name <your.email@example.com>_**

Then with a one-time command, contributors can make sure that each commit is automatically signed.
The command

``
git config commit.template ~/dscr-for-vmware/git-template
``

**Comment by rockaut** [comment]
i support that too.

**Comment by SimeonGerginov** [comment]
Yes, I also think it is a good idea to have it in the project. We will start working on it.

**Comment by SimeonGerginov** [comment]
Could you please provide more information on the requested template - the template should contain only the line specified above or to be like this for example:

```
 git add .
 git commit -s -m "<commit message>"
 git push origin <branch>
```

**Comment by lucdekens** [comment]
I was thinking of adding the following to the .gitconfig file

```
[commit]
    template = ~/.gitmessage
```

And then in ~/.gitmessage

```
Signed-off-by: Your Name <email@addre.ss>
```

**Comment by SimeonGerginov** [comment]
Ok, looks good this way !

**Comment by SimeonGerginov** [comment]
So the content of ~/.gitmessage should look like this ? 

```
 Signed-off-by: git config --global user.name git config --global user.email
```

**Comment by lucdekens** [comment]
There are a couple of ways of doing this.
1) When you are using VSC it is a preference, nothing required in git.

```
{
    "git.alwaysSignOff": true
}
```

2) When you are not using VSC, you can use a git hook, which can be a PowerShell script.
These 2 files go into the **.git\hooks** folder.

The .sh to call PowerShell, named **prepare-commit-msg**
```
#!/bin/sh
c:/Windows/System32/WindowsPowerShell/v1.0/powershell.exe -ExecutionPolicy Bypass -Command '.\.git\hooks\prepare-commit-msg.ps1'
```

and a .ps1 script, named **prepare-commit-msg.ps1**.

```
$commitText = @'
<Your commit title>

<Your commit details>

Date: $(Get-Date -Format 'ddd dd/MM/yyyy')
Time: $(Get-Date -Format 'HH:mm')

Signed-off-by: $($env:GIT_AUTHOR_NAME) <$($env:GIT_AUTHOR_EMAIL)>
'@

Set-Content -Path .\.git\COMMIT_EDITMSG -Value ($ExecutionContext.InvokeCommand.ExpandString($commitText))
```

3) Also in git, you can use the git config option **commit.template**. This config option points to a template file. Both files are located in the **.git** folder in the repo. 

File **.git\config**

```
[commit]
	template = ./.git/templatefile.txt
```

The template itself, named **templatefile.txt** here. This contains static text (I have not found how to introduce git environment variables in here).

```
<Enter commit message>

<Enter commit details>

Signed-off-by: LucD <dekens.luc@gmail.com>
```

In summary, option 1) is the easiest, but requires you use VSC.
Option 2) and 3) are editor independent, where I personally prefer option 2) since it can be customised.

**Comment by SimeonGerginov** [comment]
Yes option 2 looks very good, so I think this will be the approach we take.

**Comment by SimeonGerginov** [comment]
Another solution to the problem is to use the following template:

```
git add .
git commit -s -m <Your commit message>
git push origin <Your working branch>
```
We can add the template to the repository for reference and this way we can avoid using hooks.
What do you think ?

**Comment by lucdekens** [comment]
Since these templates will in any case be ignored for commits, I can live
with that.
Just make sure to document how it shall be used, in a language the non-git
savy user understands 😁

**Comment by SimeonGerginov** [comment]
Ok then. I will soon add it to the repository with instructions on how to use it.

**Comment by SimeonGerginov** [comment]
Added the git template to the repository. Closing the issue.
